libs/platform/src/lib/components/form/form-group/form-group.component.ts

```diff
@@ -348,126 +418,43 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
-            const merged: GroupField[] = [];
-            let indexL = 0,
-                indexR = 0,
-                current = 0;
-
-            while (current < left.length + right.length) {
-                if (indexL < left.length) {
-                    const f = new GroupField(
-                        left[indexL].zone,
-                        left[indexL].name,
-                        current,
-                        left[indexL].renderer,
-                        left[indexL].columns,
-                        left[indexL].isFluid
-                    );
-
-                    merged[current++] = f;
-                    indexL++;
-
-                    if (f.isFluid) {
-                        continue;
-                    }
-                    f.styleClass = `col-sm-${f.columns} col-md-${f.columns} col-lg-${f.columns}`;
-                }
-
-                if (indexR < right.length) {
-                    if (right[indexR].columns + left[indexL - 1].columns !== 12) {
-                        right[indexR].columns = 12 - left[indexL - 1].columns;
-                    }
-                    const f = new GroupField(
-                        right[indexR].zone,
-                        right[indexR].name,
-                        current,
-                        right[indexR].renderer,
-                        right[indexR].columns
-                    );
-
-                    f.styleClass = `col-sm-${f.columns} col-md-${f.columns} col-lg-${f.columns}`;
-                    merged[current++] = f;
-                    indexR++;
-                }
-            }
-            return merged;
-        } else if (left.length > 0) {
-            // when only one column dont use 6 columsn and 6 colums
-            return left.map((item) => {
-                item.isFluid = true;
-                return item;
-            });
-        }
-        return [];
-    }
-
-    /**
-     * This is just the temp solution until I can figure out better way of doing this. If I want
-     * to have e.g. 3 fields on the left and 1 field on the right. It will not work with current
-     * 6-column width layout as left and right side are not even.
-     *
+     * @hidden
+     * if `columnLayoutType` is given, set those column layouts appropriately
      */
-    private evenFields(zLeft: GroupField[], zRight: GroupField[]): void {
-        zLeft.sort((a, b) => a.rank - b.rank);
-
-        if (zRight.length === 0) {
-            return;
-        }
-
-        zRight.sort((a, b) => a.rank - b.rank);
-        if (zLeft.length !== zRight.length) {
-            // retrieve the smallest from both
-            const toEven = zLeft.length > zRight.length ? zRight : zLeft;
-            for (let i = 0; i <= Math.abs(zLeft.length - zRight.length); i++) {
-                const zone = toEven[0].zone;
-                toEven.push(new GroupField(zone, `${zone}-${i}`, toEven.length + 1, null, 6));
+    private _setUserSpecifiedLayout(groupField?: GroupField): void {
```

Comment:
We need to also validate highest column numbers with our layout pattern. 

So will not have XL2, but you will try to place column into number 4.

Reply:
I don't think so we need to validate this case. At now if a form has 2 columns and a field will try bind to the 4th column, it will set at the last available column